app/models/ems_event.rb

```diff
@@ -76,10 +76,6 @@ def self.add_openstack_infra(ems_id, event)
     add(ems_id, ManageIQ::Providers::Openstack::InfraManager::EventParser.event_to_hash(event, ems_id))
   end
 
-  def self.add_amazon(ems_id, event)
-    add(ems_id, ManageIQ::Providers::Amazon::CloudManager::EventParser.event_to_hash(event, ems_id))
-  end
```

Comment:
I think this introduces a coupling of `EmsEvent` to a `EventParser` in a class. But I like the idea of moving the `add` out of the Amazon runner.